charts/signoz/templates/schema-migrator/migrations-sync.yaml

```diff
@@ -13,17 +13,20 @@ metadata:
     {{- with .Values.schemaMigrator.annotations }}
     {{- toYaml . | nindent 4 }}
     {{- end }}
-    {{- if and .Release.IsUpgrade .Values.schemaMigrator.upgradeHelmHooks }}
+    {{- if .Values.schemaMigrator.upgradeHelmHooks }}
+    argocd.argoproj.io/hook: Sync
+    argocd.argoproj.io/hook-delete-policy: BeforeHookCreation
+    {{- if .Release.IsUpgrade }}
     helm.sh/hook: pre-upgrade
     helm.sh/hook-delete-policy: before-hook-creation
     {{- end }}
+    {{- end }}
```

Comment:
> migrations-sync.yaml uses pre-upgrade Helm hook with Sync ArgoCD hook

Since every sync in ArgoCD is applied using Helm install generated templates, we use the `Sync` hook to keep the behaviour consistent and not break the K8s Job cleanup.